build_ipa_sles.sh

```diff
@@ -34,57 +47,65 @@ QUIET_CLEANUP="${QUIET_CLEANUP:-false}"
     rm -rf "$IPA_BUILD_WORKSPACE"
 fi
 
-# Install required packages
-#sudo apt install --yes python3-pip python3-virtualenv qemu-utils
-dnf install -y python3-pip python3-virtualenv qemu-utils
+#################################################
+# Install required packages and setup host
+#################################################
+if [[ "${ID}" == "ubuntu" ]]; then
+    sudo apt install -y python3-pip qemu-utils
+elif [[ "${ID}" == "centos" ]]; then
+    sudo dnf install -y python3-pip qemu-img
+fi
 
 # Create the work directory
 mkdir --parents "${IPA_BUILD_WORKSPACE}"
 cd "${IPA_BUILD_WORKSPACE}"
 
 # Install the cloned IPA builder tool
-virtualenv venv
+python3 -m venv venv
 # shellcheck source=/dev/null
 source "./venv/bin/activate"
 python3 -m pip install --upgrade pip
 python3 -m pip install "diskimage-builder"
 
+#################################################
 # Export variables that will be used by DIB
-## Configure the IPA builder to pull the IPA source from Nordix fork
+#################################################
+# Configure the IPA builder to pull the IPA source from Nordix fork
 export DIB_REPOLOCATION_ironic_python_agent="${IPA_REPO}"
 export DIB_REPOLOCATION_ironic_lib="${IRONIC_LIB_REPO}"
 export DIB_REPOLOCATION_requirements="${OPENSTACK_REQUIREMENTS_REPO}"
 export DIB_REPOREF_requirements="${OPENSTACK_REQUIREMENTS_REF}"
 export DIB_REPOREF_ironic_python_agent="${IPA_REF}"
 export DIB_REPOREF_ironic_lib="${IRONIC_LIB_REF}"
 export DIB_DEV_USER_USERNAME=metal3
-export DIB_REPOLOCATION_glean="https://github.com/Nordix/glean.git"
-export DIB_REPOREF_glean="refs/heads/parsing_error"
+export DIB_INSTALLTYPE_pip_and_virtualenv="package"
+
 if [ "${ENABLE_DEV_USER_PASS}" == "true" ]; then
 export DIB_DEV_USER_PWDLESS_SUDO=yes
 export DIB_DEV_USER_PASSWORD="test"
 fi
 if [ "${ENABLE_DEV_USER_SSH}" == "true" ]; then
 export DIB_DEV_USER_AUTHORIZED_KEYS="${DEV_USER_SSH_PATH}"
 fi
-export DIB_INSTALLTYPE_pip_and_virtualenv="package"
-## List of additional kernel modules that should be loaded during boot separated by space
-## This list is used by the custom element named ipa-modprobe
+
+# List of additional kernel modules that should be loaded during boot separated by space
+# This list is used by the custom element named ipa-modprobe
 export DIB_ADDITIONAL_IPA_KERNEL_MODULES="megaraid_sas"
 ## Provide path(s) of the custom elemnts for DIB
 CUSTOM_ELEMENTS="${CURRENT_SCRIPT_DIR}/ipa_builder_elements"
 export ELEMENTS_PATH="${ELEMENTS_PATH:-${CUSTOM_ELEMENTS}}"
 
+#################################################
 # Build the IPA initramfs and kernel images
+#################################################
 disk-image-create \
     "sles-ipa-install" "${IPA_BASE_OS}" "sles-zypper-config" "sles-ipa-ramdisk-base" \
     "dynamic-login" "journal-to-console" "devuser" "openssh-server" "sles-extra-hardware" \
-    "ipa-module-autoload" "simple-init" "override-simple-init" -o "${IPA_IMAGE_NAME}"
```

Comment:
Make a condition please where simple-init is installed, simple init is still a useful thing we just don't need it usually.